File: ecmascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

class Bob {
  hey (message) {
    let reply = 'Whatever.'
    if (message.endsWith('?')) {
      reply = 'Sure.'
    }
    if (/[A-z]/.test(message) && message.toUpperCase() === message) {
      reply = 'Whoa, chill out!'
    }
    if (message.trim() === '') {
      reply = 'Fine. Be that way!'
    }
    return reply
  }
}

export default Bob

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.ends_with? input, "?" ->
        "Sure."
      String.trim(input) == "" ->
        "Fine. Be that way!"
      String.upcase(input) == input && String.match?(input, ~r/[[:alpha:]]/)->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end
end
